server/src/database/archive.rs

```diff
@@ -90,3 +91,88 @@ pub(crate) fn fetch_transactions(
         .get_results(connection)?;
     Ok(results)
 }
+
+use std::collections::HashMap;
+
+use graphql_client::{reqwest::post_graphql_blocking as post_graphql, GraphQLQuery, Response};
+use reqwest::blocking::Client;
+
+use transaction_query::TransactionQueryTransactions;
+
+type DateTime = String;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "src/database/transaction_schema.graphql",
+    query_path = "src/database/transaction_query.graphql",
+    response_derives = "Debug"
+)]
+pub struct TransactionQuery;
+
+#[allow(clippy::unwrap_used, clippy::upper_case_acronyms)]
+pub(crate) fn fetch_transactions_graphql(
+    start_time_millis: i64,
+    end_time_millis: i64,
+    mip_key: &str,
+    base_memo: &str, // Pass the base memo as an argument
+) -> Result<Vec<FetchTransactionResult>> {
+    let start_unix_timestamp = start_time_millis / 1000; // Convert milliseconds to seconds
+    let end_unix_timestamp = end_time_millis / 1000;     // Convert milliseconds to seconds
+
+    let start_datetime = OffsetDateTime::from_unix_timestamp(start_unix_timestamp);
+    let end_datetime = OffsetDateTime::from_unix_timestamp(end_unix_timestamp);
```

Comment:
There will be a loss of precision (though it might not matter). I would find the millis equivalent to create your datetime objects

Reply:
addressed in recent commit below https://github.com/Granola-Team/mina-on-chain-voting/pull/237/commits/23f1e63838de67559b22f350517f91e2ac127570